File: backend/internal/model/product.go
package model

import "time"

// Product 商品实体：Images 为 JSON 数组字符串，Detail 为富文本。
type Product struct {
	ID            uint      `gorm:"primaryKey" json:"id"`
	CategoryID    uint      `gorm:"index;not null" json:"category_id"`
	Name          string    `gorm:"size:128;not null;index" json:"name"`
	SubTitle      string    `gorm:"size:255" json:"sub_title"`
	Description   string    `gorm:"type:text" json:"description"`
	Detail        string    `gorm:"type:text" json:"detail"`
	CoverImage    string    `gorm:"size:255" json:"cover_image"`
	Images        string    `gorm:"type:text" json:"images"`
	Price         float64   `gorm:"type:decimal(10,2);not null" json:"price"`
	OriginalPrice float64   `gorm:"type:decimal(10,2)" json:"original_price"`
	Stock         int       `gorm:"not null;default:0" json:"stock"`
	Sales         int       `gorm:"default:0" json:"sales"`
	Rating        float64   `gorm:"type:decimal(3,2);default:5" json:"rating"`
	RatingCount   int       `gorm:"default:0" json:"rating_count"`
	ShippingFrom  string    `gorm:"size:64" json:"shipping_from"`
	FreeShipping  bool      `gorm:"default:false" json:"free_shipping"`
	Status        string    `gorm:"size:16;default:ON_SALE;index" json:"status"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

File: frontend/src/api/product.ts
import { get, post, put } from '../utils/request';

export interface ProductVO {
  id: number;
  category_id: number;
  category_name: string;
  name: string;
  sub_title: string;
  description: string;
  detail: string;
  cover_image: string;
  images: string[];
  price: number;
  original_price: number;
  stock: number;
  sales: number;
  rating: number;
  rating_count: number;
  shipping_from: string;
  free_shipping: boolean;
  status: string;
  created_at: string;
}

export interface PageData<T> {
  list: T[];
  total: number;
  page: number;
  page_size: number;
}

export interface ProductQuery {
  page?: number;
  page_size?: number;
  keyword?: string;
  category_id?: number;
  min_price?: number;
  max_price?: number;
  free_shipping?: boolean;
  shipping_from?: string;
  sort?: string;
  status?: string;
}

export const listProducts = (params?: ProductQuery) => get<PageData<ProductVO>>('/products', params as Record<string, unknown>);

export const getProduct = (id: number) => get<ProductVO>(`/products/${id}`);

export const recordView = (id: number) => post<{ message: string }>(`/products/${id}/view`);

export const getRecommendations = () => get<ProductVO[]>('/products/recommendations');
